eng/targets/ResolveReferences.targets

```diff
@@ -47,19 +47,18 @@
       Projects should only use the project references instead of baseline package references when:
```

Comment:
Changes here
1. copy over latest fixes from aspnet/AspNetCore
2. does new stuff to handle the new ref/-dependent compilation requirement
3. addresses @wtgodbe's feedback

Please categorize because I'd like to focus on (2).